sys-fs/btrfsmaintenance/btrfsmaintenance-0.3.1.ebuild

```diff
@@ -0,0 +1,45 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+inherit eutils
+
+DESCRIPTION="Scripts for btrfs maintenance tasks like periodic scrub, balance, trim or defrag"
+HOMEPAGE="https://github.com/kdave/btrfsmaintenance"
+SRC_URI="https://github.com/kdave/btrfsmaintenance/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND=""
+RDEPEND="${DEPEND}
+	sys-fs/btrfs-progs
+	virtual/cron"
+
+src_install() {
+	dodoc README.md CONTRIBUTING.md btrfsmaintenance.changes
+	insinto /etc/default
+	newins sysconfig.btrfsmaintenance btrfsmaintenance
+	insinto /usr/share/btrfsmaintenance
+	doins btrfsmaintenance-functions
+	exeinto /usr/share/btrfsmaintenance
+	doexe btrfs*.sh
+	insinto /usr/lib/systemd/system
+	doins btrfsmaintenance-refresh.service
+}
+
+pkg_postinst() {
+	elog "Installing default btrfsmaintenance scripts"
+	"${EROOT%/}"/usr/share/btrfsmaintenance/btrfsmaintenance-refresh-cron.sh || die
+	elog "Now edit cron periods and mount points in /etc/default/btrfsmaintenance"
+	elog "then run /usr/share/btrfsmaintenance/btrfsmaintenance-refresh-cron.sh to"
+	elog "update cron symlinks"
+}
+
+pkg_prerm() {
+	# uninstall symlinks
+	elog "Removing symlinks from btrfsmaintenance cron tasks"
+	"${EROOT%/}"/usr/share/btrfsmaintenance/btrfsmaintenance-refresh-cron.sh uninstall || die
```

Comment:
Here is a emerge log from an upgrade:

> >>> Installing (1 of 1) sys-fs/btrfsmaintenance-0.3.1::md-private
>  * Removing symlinks from btrfsmaintenance cron tasks
>  * Installing default btrfsmaintenance scripts
> Refresh script btrfs-scrub.sh for monthly
> Refresh script btrfs-defrag.sh for none
> Refresh script btrfs-balance.sh for weekly
> Refresh script btrfs-trim.sh for weekly
>  * Now edit cron periods and mount points in /etc/default/btrfsmaintenance
>  * then run /usr/share/btrfsmaintenance/btrfsmaintenance-refresh-cron.sh to
>  * update cron symlinks
> >>> Auto-cleaning packages...
> 
> >>> No outdated packages were found on your system.
> 

So on upgrade the symlinks are first removed, then created again - which gives a perfect result (the symlinks exist) with a small work overhead (remove usually not necessary). Do you know a good way to determine in the pkg_prerm function if the remove is final or before an upgrade? 
But I am in doubt if this makes sense, because I figured out that the pkg_prerm function is called from the OLD ebuild (which is going to remove) and afterwards the pkg_postinstall is called from the NEW ebuild. So the logic is: the old ebuild cleans everything, regardless of a remove or upgrade situation. I am fine with this.